src/combinator.ts

```diff
@@ -29,6 +29,24 @@ export const arrayP = <T>(type: TypeProxy<T>): TypeProxy<T[]> => (value) => {
 
 export const defaultP = <T>(defaultValue: T, type: TypeProxy<T>): TypeProxy<T> => or2P(type, pureP(defaultValue));
 
+export const jsonP = <T>(type: TypeProxy<T>): TypeProxy<T> => (value) => {
+  if (typeof value !== 'string') {
+    return {
+      error: ParseError.simpleError(value, 'JSON string'),
+      success: false
+    };
+  }
+
+  try {
+    return type(JSON.parse(value));
+  } catch (error) {
+    return {
+      error: ParseError.simpleError(value, 'valid JSON'),
+      success: false,
+    };
+  }
```

Comment:
If I'm being pedantic, the correct way to do this functionally is to leave `jsonP` as `jsonP: TypeProxy<unknown>` (i.e don't take a type as an argument), and then implement `ComposeP: <A, B>(type1: TypeProxy<A>, type2: TypeProxy<B>) => TypeProxy<B>`. Then you would use `composeP(jsonP, type)`

Reply:
:nice: